Helm wip

Sedward opened the pull request:
Here is one more pull request to get the persistent volume claim working and some aesthetic changes based on some issues I ran into when naming specs. Again, maybe not everything gets merged. If there are any questions, let me know. 

At this point we should consider using github issues to identify what changes to make and assign them to contributors. That way we aren't stepping on each others toes with commit. In addition, we can discuss certain changes before actually making them and ensure everyone agrees on the proposed changes.  I'd even go as far as saying we could maybe move this the charts/incubator or it's own temporary repo.

mumoshu committed:
Merge pull request #14 from vyper/patch-1

Typo on README

gsemet committed:
Add Helm chart for Airflow

- add EditorConfig files
- airflow.all.yaml is splited in different chart templates
- Propagate server config from env variable if defined
- statup script can receive some config by ENV variable
  (name of rabbitmq/postgres server), at least everything
  derives from the same source
- username and password can inherit from env
- fix probable buggy fernet key definition, missing enum34
  with python 2 installation
- change format of template in airflow.cfg (airflow.cfg.in)

Signed-off-by: Gaetan Semet <gaetan@xeberon.net>

Commit:
This commit adds and icon to the Chart.
This is not required but still shows up in helm lint tests

Commit:
This commit adds a helpers to access name variables in the templates

Commit:
Adding of optional pvc WIP
This commit adds an optional PVC for the dags directory.

Commit:
Refactor deployments and add a dags path variable
This commit breaks the deployments out into separate files. The purpose
of this is for troubleshooting when making changes to the deployment and
fixing errors when finding which deployment and line number is causing
a problem. This change also make the dag_path which is used for
volumeMounts within deployments.

Commit:
Change the PVC access mode
Ths changes the acces mode to ReadWriteMany because it will be accessed
more than one pod

Commit:
This commit adds a more up to date version
of the pvc and values to go with the persistence section

Commit:
Merge branch 'helm_chart' of github.com:Stibbons/kube-airflow

Commit:
This commit is a post-merge conflict cleanup commit.
* removes the duplicate deployment files in favor of the
files prefixed with deployment_
* fixes a renamed variable in templates/deployments-worker.yaml

Commit:
This commit make changes to many spec names
1. Remove the prefix variable and populate it with the helper fullname.
This way, it will autopopulate with a legal name, but can be overidden
by the helm user with a legal name from the command line.
2. Change the port names to be hardcoded. This is to keep them under the
15 character limit and not break a deploy.

Commit:
This commit changes the persistent volume claims
1. Allows the persistent volume claim to be named explicitly
2. Creates a separate claim for the worker and scheduler

gsemet commented:
Can you rebase on my branch? I did a few hard push.

and I agree with tickets, please open them !

gsemet commented on airflow/templates/configmaps.yaml:
```diff
@@ -1,7 +1,7 @@
 kind: ConfigMap
 metadata:
-  name: "{{ .Values.airflow.prefix -}}airflow-cfg"
+  name: {{ template "airflow.fullname" . }}-airflow-cfg
```
I don't exactly see the point?

Don't you want something like
```
{{ template "airflow.fullname" airflow-cfg }}
```

Sedward replied:
In the docs they have something like this

apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ .Release.Name }}-configmap

I was mostly trying  to conform to this. I haven't tried this syntax but I can.

gsemet replied:
ok i'll see what this do tomorow. My requirement is to be able to change this release easily so we can have several instance of airflow in the same namespace (especially, we share the namespace with another celery+rabbitmq), but we are also trying kubernetes executor fork, all in the same namespace

gsemet commented on airflow/templates/services.yaml:
```diff
@@ -1,67 +1,67 @@
     tier: rabbitmq
   ports:
-    - name: {{ .Values.airflow.prefix -}}node
+    - name: rabbitmq
```
please keep the prefix, I need it to allow several airflow in the same namespace

Sedward replied:
I understand. The issue I ran into what the Release Name was somewhat long and combined with the service, it surpassed the 15 character limit.

gsemet replied:
yes it needs to be small and dns compatible

gsemet commented on airflow/values.yaml:
```diff
@@ -43,14 +40,16 @@ db:
   ##   GKE, AWS & OpenStack)
   # storageClass: default
+  # If defined, will overide the claim name
+  claimName: "killjoy-dragonfly"
```
should be in your config.yaml

gsemet commented on airflow/values.yaml:
```diff
@@ -43,14 +40,16 @@ db:
 
 persistence:
-  enabled: false
+  enabled: true
```
set to false by default, user will enable it in their config.yaml

Sedward replied:
agreed. that was a mistake

gsemet commented on reset-af.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
I prefere using the makefile. works great

Sedward commented:
Making fixes. I pushed too hastily.

gsemet commented:
Sorry I pushed other commits, please rebase

Commit:
This commit fixes some values that should not be default
1. persistent volumes should default to false
2. The persistent volume claim name should not be set

Commit:
reset-af.sh is not needed

Sedward commented:
I think we need to enable issues for this repo? I don't see the issue at the top.

gsemet commented:
Done !
